documentation/development-docs/azure-powershell-developer-guide.md

```diff
@@ -415,6 +415,22 @@ using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 public string ResourceGroupName { get; set; }
 ```
 
+### Resource Name Completer
+
+For any parameter that takes a resource name, the `ResourceNameCompleter` should be applied as an attribute.  This will allow the user to tab through all resource names for the ResourceType in the current subscription.  This completer will filter based upon the current parent resources provided (for instance, if ResourceGroupName is provided, only the resources in that particular resource group will be returned).  For this completer, please provide the ResourceType as the first argument, followed by the parameter name for all parent resources starting at the top level.
+
+```cs
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+...
+[Parameter(Mandatory = false, HelpMessage = "The parent server name")]
+[ResourceNameCompleter("Microsoft.Sql/servers", "ResourceGroupName")]
+public string ServerName { get; set; }
+
+[Parameter(Mandatory = false, HelpMessage = "The database name")]
+[ResourceNameCompleter("Microsoft.Sql/servers/databases", "ResourceGroupName", "ServerName")]
```

Comment:
I think you can use ```nameof(ResourcegroupName)``` here instead of ```"ResourceGroupName"```.  This will make these easier to review, and also ensure that any name change is always reflected.

Reply:
Also, I think we should make this part of the requirements in the design guidelines.  Also, I think we should write up a blog post on this new feature.